routingpy/utils.py

```diff
@@ -81,14 +81,28 @@ def decode_polyline5(polyline, is3d=False):
         support this. Default False.
     :type is3d: bool
 
+    :param order: Specifies the order in which the coordinates are returned.
+                  Options: latlng, lnglat. Defaults to 'lnglat'.
+    :type order: str
+
     :returns: List of decoded coordinates with precision 5.
     :rtype: list
     """
-
-    return _decode(polyline, precision=5, is3d=is3d)
+    if order == "latlng":
+        return _decode(polyline, precision=5, is3d=is3d)
+    elif order == "lnglat":
+        if is3d:
+            return [
+                tuple((reversed(coords[:-1]), coords[-1]))
+                for coords in _decode(polyline, precision=5, is3d=is3d)
+            ]
+        else:
+            return [tuple(reversed(coords)) for coords in _decode(polyline, precision=5, is3d=is3d)]
+    else:
+        raise ValueError(f"order must be either 'latlng' or 'lnglat', not {order}.")
```

Comment:
I think this makes more sense in the `_decode` function. 

suggestion:
1. pass the `order` to `_decode`
2. a small lambda in `_decode` to give you the right order, e.g.

```
get_coords = lambda lng, lat, factor: (lat / factor, lng / factor) if order == "latlng" else (lng / factor, lat / factor)

# then when building coords
coordinates.append((*get_coords(lng, lat, factor), z / 100))
```

you can also pull the check if the right order choice was passed into `_decode` IMO.

Reply:
not entirely sure about scope here actually.. maybe better to rename the lambda args to `x, y, f` so we don't shadow

Reply:
also I didn't test the suggestion ;)